internal/cfgplugins/isis.go

```diff
@@ -22,6 +23,79 @@ type ISISConfigBasic struct {
 	LoopbackIntf string
 }
 
+// ISISGlobalParams is the data structure for the DUT data.
+type ISISGlobalParams struct {
+	DUTArea             string
+	DUTSysID            string
+	NetworkInstanceName string
+	ISISInterfaceNames  []string
+}
+
+// NewISIS configures the DUT with ISIS protocol.
+func NewISIS(t *testing.T, dut *ondatra.DUTDevice, ISISData *ISISGlobalParams) *oc.Root {
```

Comment:
NewISIS should have a batch object as a parameter.

The caller should create the batch object and pass it to one or more cfgplugins, then call gnmi.Set(). (same style as cfgplugins.NewAggregateInterface)